README.md

```diff
@@ -44,59 +50,53 @@ cd generative_data_prep
+|   `pretrained_tokenizer`    |  str | The tokenizer to use when tokenizing the input dataset. The tokenizers are model specific and can be found on HuggingFace.      |  You can get this value by copying the model path available on the HuggingFace model card. I.e. For Llama-2-7b-hf I would put `"meta-llama/Llama-2-7b-hf"`  |
+|   `max_seq_length`          |  int | The max size of input sequence a model can support. This is model specific - i.e. for `GPT-2` it's __1024__, for `Llama-2` it's __4096__.  |   You can find this information in a few places, but a place you can consistently find this value is in the `config.json` file under the HuffingFace model's File's and Versions tab. Then grab the value for `max_position_embeddings`.  |
+|   `input_packing_config`    |  str | Defines the strategy used to pack the provided data into sequences across the output HDF5 files. |   There are 7 options for this flag: `'full'`, `'single::truncate_left'`, `'single::truncate_right'`, `'single::drop'`, `'greedy::truncate_left'`, `'greedy::truncate_right'`, `'greedy::drop'`. Check out the [`input_packing_config` flag](#input_packing_config) below for an in depth description of these options. |
+|          `shuffle`          |  int | Determines whether to shuffle the input dataset, and whether to shuffle on RAM. |   Check out the [shuffle flag](#shuffle) below for more details.  |
 
-Each line in the input file must be either plain text or [jsonline](https://jsonlines.org/). if the jsonline has different keywords, refer to the `prompt_keyword`, and `completion_keyword` flag documentation below.
+### Example
+```shell
+python3 -m generative_data_prep pipeline --input_file_path=<PATH TO DATASET FILE> --output_path=<PATH TO OUTPUT DIRECTORY> --pretrained_tokenizer=gpt2 --max_seq_length=1024 --input_packing_config='single::drop' --shuffle=on_RAM
+```
 
-Each line in the input file can be formatted as one of the following:
-- Plain text
-- `{"prompt": "", "completion": ""}`
-- `[{"prompt": "text...", "completion": "text..."}, {"prompt": "text...", "completion": "text..."}, {"prompt": "text...", "completion": "text..."}, ...]`
+</br>
 
+## Input
 
-## End to end data preparation
-The `generative_data_prep/data_prep/pipeline.py` script takes a single jsonline or text file as input, shuffles it, splits it into multiple train/dev/test, tokenizes the text, packs it into fixed sequence lengths and then converts it to HDF5 file format.
+The input file format must be either `.txt` or [`.jsonl`](https://jsonlines.org/). Depending on whether you want to do [fine-tuning](#fine-tuning) or [pretraining](#pretraining), your data will need to be in a specific format.
```

Comment:
This implies to me that .txt or .jsonl impacts fine tuning or pre-training. 

the second sentence is not very informative, maybe remove?